### DISCUSSION TITLE
Detect redundant config

### DISCUSSION THREAD

**Discussion by vb** [eslint/eslint#15358] [Ideas] [answered]
Sometimes you work on a project where a .eslintrc file has lived for ages. Lots of plugins, extensions, overrides, and rules.

I was looking at such a file and wondered, how can I remove redundant config? Config that is overridden by other extensions/plugins/rules. 

I found --print-config flag for the CLI helpful. I can see the "final", or merged config that eslint uses.
However, I feel like it would be even more helpful to get some information on why a certain rule or config is present.

Simplified Example

```
// .eslintrc
{
  extends: [a, b, c]
  rules: {
    ruleA: "off",
    ruleC: "warn"
    ...
  }
}
```

--print-config with comments on origin, overrides, etc
```
{
  rules: {
    ruleA: "off" // Added by .eslintrc, a
    ruleBC: "off" // Added by b and a
    // etc...
  }
}
```

Such a feature would help in debugging and refactoring config files!

**Comment by nzakas** [comment, marked as answer]
That’s an interesting idea, but we aren’t making any further changes to the current configuration system as we work on replacing it. The new config system uses regular import/require statements, so you easily figure out where rules come from.
